ee/billing/quota_limiting.py

```diff
@@ -80,21 +105,27 @@ def remove_limited_team_tokens(resource: QuotaResource, tokens: list[str], cache
 
 @cache_for(timedelta(seconds=30), background_refresh=True)
 def list_limited_team_attributes(resource: QuotaResource, cache_key: QuotaLimitingCaches) -> list[str]:
+    """
+    Returns a list of team attributes that are still under quota limits. Uses Redis sorted set
+    where scores are expiration timestamps. Only returns attributes whose limits haven't expired yet.
+    Note: this is cached for 30 seconds so it's not always up to date.
+    """
     now = timezone.now()
     redis_client = get_client()
     results = redis_client.zrangebyscore(f"{cache_key.value}{resource.value}", min=now.timestamp(), max="+inf")
     return [x.decode("utf-8") for x in results]
 
 
-class UsageCounters(TypedDict):
-    events: int
-    recordings: int
-    rows_synced: int
+# -------------------------------------------------------------------------------------------------
+# MAIN FUNCTIONS
+# -------------------------------------------------------------------------------------------------
 
 
 def org_quota_limited_until(
     organization: Organization, resource: QuotaResource, previously_quota_limited_team_tokens: list[str]
 ) -> Optional[OrgQuotaLimitingInformation]:
+    report_organization_action(organization, "quota limiting check started", properties={"resource": resource.value})
```

Comment:
this could be a large volume of events since this runs every 30 mins - does it only run this function if the org has a limit on it?

Reply:
yeah you're right, I can try and reduce the number of calls I make. Trying to balance insight vs not over doing it.

Reply:
limited it more and most will be ignore from not hitting limits - I'll only leave it live for a week or so as I debug